src/js/messaging/containers/Thread.jsx

```diff
@@ -69,19 +86,19 @@ class Thread extends React.Component {
     const thread = this.props.thread;
     const folderMessages = this.props.folderMessages;
     const folderMessageCount = folderMessages.length;
+
     let lastSender;
     let header;
     let threadMessages;
+    let currentMessage;
 
     // Exclude the current folder from the list of folders
     // that are passed down to the MoveTo component.
     const folders = this.props.folders.filter((folder) => {
       return folder.folderId !== this.props.persistFolder && folder.name !== 'Sent';
     });
 
-    if (thread.length > 0) {
-      const currentMessage = thread[thread.length - 1];
-
+    if (this.props.message) {
       // TODO: Presumably, when the API provides pagination,
```

Comment:
No, I guess not. Pagination is in, but it doesn't support links to the next or previous message. But thinking about it, I'm not sure it can. I think we'll have to handle the case ourselves where you're on the last message of the page and want to go next (and conversely, going to the previous if you're on the first message of a page after the first page).